addons/stock/static/src/js/forecast_widget.js

```diff
@@ -56,14 +56,14 @@ const ForecastWidgetField = AbstractField.extend({
         const productId = this.recordData.product_id.res_id;
         const resModel = 'product.product';
         this._rpc({
-            model: resModel,
+            model: 'stock.move',
             method: 'action_product_forecast_report',
-            args: [productId],
+            args: [this.recordData.id],
         }).then(action => {
-            action.context = {
+            action.context = Object.assign(action.context || {}, {
                 active_model: resModel,
```

Comment:
The `resModel` variable is useless if you use it only in one place. You can remove it and write directly `product.product` instead.